lib/script.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# implementing the player class
+class Player
+  attr_accessor :score
+  attr_accessor :choice
+  attr_reader :name
+  def initialize(name)
+    @choice = nil
+    @name = name
+    @score = 0
+  end
+end
+# implementing the Board class
+class Board
+  attr_writer :matrix
+  def initialize
+    @counter = 0
+    @matrix = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+  end
+
+  def update(choice, place)
+    if place <= 3
+      @matrix[0][@matrix[0].index(place)] = choice
+    elsif place <= 6
+      @matrix[1][@matrix[1].index(place)] = choice
+    else
+      @matrix[2][@matrix[2].index(place)] = choice
+    end
+    display
+    result = Result.new(@matrix)
+    result.check_success
+    @counter += 1
+    [result.check_success, @counter]
+  end
+
+  def display
```

Comment:
should not mix user interaction with game logic (e.g. no “puts” and “get.chomp” in lib directory!).move it to the bin.